dvc/_debug.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+from contextlib import ExitStack, contextmanager
+
+
+@contextmanager
+def instrument(html_output):
+    """Run a statistical profiler"""
+    try:
+        from pyinstrument import Profiler
+    except ImportError:
+        print("Failed to run profiler, pyinstrument is not installed")
+        yield
+        return
+
+    profiler = Profiler()
+
+    profiler.start()
+    yield
+    profiler.stop()
+
+    if html_output:
+        profiler.open_in_browser()
+        return
+    print(profiler.output_text(unicode=True, color=True))
+
+
+@contextmanager
+def profile(dump):
+    """Run a cprofile"""
+    import cProfile
+
+    prof = cProfile.Profile()
+    prof.enable()
+
+    yield
+
+    prof.disable()
+    if not dump:
+        prof.print_stats(sort="cumtime")
+        return
+    prof.dump_stats(dump)
+
+
+@contextmanager
+def debug():
+    try:
+        yield
+    except Exception:  # pylint: disable=broad-except
+        import pdb  # noqa: T100
```

Comment:
Maybe use `ipdb` then fallback to `pdb`?

Reply:
@skshetry makes sense!